src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -41,19 +43,20 @@ public static IOperation Create(HttpPipeline pipeline, RequestMethod requestMeth
                     ? (true, locationUri)
                     : (false, null);
 
-            return new NextLinkOperationImplementation(pipeline, requestMethod, startRequestUri, nextRequestUri, headerSource, originalResponseHasLocation, lastKnownLocation, finalStateVia);
+            return new NextLinkOperationImplementation(pipeline, request, nextRequestUri, headerSource, originalResponseHasLocation, lastKnownLocation, finalStateVia);
         }
 
-        private NextLinkOperationImplementation(HttpPipeline pipeline, RequestMethod requestMethod, Uri startRequestUri, string nextRequestUri, HeaderSource headerSource, bool originalResponseHasLocation, string? lastKnownLocation, OperationFinalStateVia finalStateVia)
+        private NextLinkOperationImplementation(HttpPipeline pipeline, Request request, string nextRequestUri, HeaderSource headerSource, bool originalResponseHasLocation, string? lastKnownLocation, OperationFinalStateVia finalStateVia)
         {
-            _requestMethod = requestMethod;
+            _requestMethod = request.Method;
             _headerSource = headerSource;
-            _startRequestUri = startRequestUri;
+            _startRequestUri = request.Uri.ToUri();
             _nextRequestUri = nextRequestUri;
             _originalResponseHasLocation = originalResponseHasLocation;
             _lastKnownLocation = lastKnownLocation;
             _finalStateVia = finalStateVia;
             _pipeline = pipeline;
+            _userAgent = request.Headers.TryGetValues("User-Agent", out IEnumerable<string>? userAgent) ? string.Join(" ", userAgent) : default;
```

Comment:
Does it not work to simply call TryGetValue and let it do the string manipulation?

Reply:
TryGetValue will do the string manipulation with comma, so the generated `User-Agent` header will like this:
```
"User-Agent": "azsdk-net-ResourceManager.Cdn/1.0.0-alpha.20220119.1,(.NET 6.0.1; Microsoft Windows 10.0.19044)"
```
But for now our 'User-Agent' will be an array and looks like this:
```
"User-Agent": [
          "azsdk-net-ResourceManager.Cdn/1.0.0-alpha.20220119.1",
          "(.NET 6.0.1; Microsoft Windows 10.0.19044)"
        ]
```
That's the reason I use TryGetValues and join the string by myself.